cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,361 @@
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"runtime"
+	"strconv"
+	"time"
+
+	googlecloud "cloud.google.com/go/storage"
+	"github.com/dgraph-io/badger/v2"
+	grpczerolog "github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/tags"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	unstaked "github.com/onflow/flow-go/follower"
+	"github.com/onflow/flow-go/model/flow"
+
+	api "github.com/optakt/flow-dps/api/dps"
+	"github.com/optakt/flow-dps/codec/zbor"
+	"github.com/optakt/flow-dps/metrics/output"
+	"github.com/optakt/flow-dps/metrics/rcrowley"
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/service/cloud"
+	"github.com/optakt/flow-dps/service/follower"
+	"github.com/optakt/flow-dps/service/forest"
+	"github.com/optakt/flow-dps/service/index"
+	"github.com/optakt/flow-dps/service/loader"
+	"github.com/optakt/flow-dps/service/mapper"
+	"github.com/optakt/flow-dps/service/metrics"
+	"github.com/optakt/flow-dps/service/storage"
+)
+
+const (
+	success = 0
+	failure = 1
+)
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+
+	// Signal catching for clean shutdown.
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+
+	// Command line parameter initialization.
+	var (
+		flagBindAddr        string
+		flagBucket          string
+		flagCheckpoint      string
+		flagData            string
+		flagForce           bool
+		flagIndex           string
+		flagLevel           string
+		flagMetrics         bool
+		flagMetricsInterval time.Duration
+		flagNodeID          string
+		flagPeerAddr        string
+		flagPeerKey         string
+		flagPort            uint16
+	)
+
+	pflag.StringVar(&flagBindAddr, "bind-addr", "127.0.0.1:5006", "address on which to bind the FIXME")
+	pflag.StringVar(&flagBucket, "bucket", "", "name of the Google Cloud Storage bucket which contains the block data")
+	pflag.StringVarP(&flagCheckpoint, "checkpoint", "c", "", "checkpoint file for state trie")
+	pflag.StringVarP(&flagData, "data", "d", "", "database directory for protocol data")
+	pflag.BoolVarP(&flagForce, "force", "f", false, "overwrite existing index database")
+	pflag.StringVarP(&flagIndex, "index", "i", "index", "database directory for state index")
+	pflag.StringVarP(&flagLevel, "level", "l", "info", "log output level")
+	pflag.BoolVarP(&flagMetrics, "metrics", "m", false, "enable metrics collection and output")
+	pflag.DurationVar(&flagMetricsInterval, "metrics-interval", 5*time.Minute, "defines the interval of metrics output to log")
```

Comment:
Defaults should really be defined right there, where the flags are defined. Otherwise, we have to start looking in two different spots of the code to know, for no obvious reason imo.